protected/models/ar/Project.php

```diff
@@ -204,60 +191,22 @@ public function getSurvey(): SurveyInterface
         return $this->limesurveyDataProvider()->getSurvey($this->base_survey_eid);
     }
 
-    public function dataSurveyOptions(): array
-    {
-        $existing = Project::find()->select('base_survey_eid')->indexBy('base_survey_eid')->column();
-
-        $surveys = filter(function ($details) use ($existing) {
-            return $this->base_survey_eid == $details['sid'] || !isset($existing[$details['sid']]);
-        }, $this->limesurveyDataProvider()->listSurveys());
-
-        $result = ArrayHelper::map($surveys, 'sid', function ($details) {
-                return $details['surveyls_title'] . (($details['active'] == 'N') ? " (INACTIVE)" : "");
-        });
-
-        return $result;
-    }
-
     public function getWorkspaces(): ActiveQuery
     {
         return $this->hasMany(Workspace::class, ['tool_id' => 'id'])->inverseOf('project');
     }
 
-    public function getTypemapAsJson(): string
-    {
-        return Json::encode($this->typemap, JSON_PRETTY_PRINT);
-    }
-
-    public function getOverridesAsJson(): string
-    {
-        return Json::encode($this->overrides, JSON_PRETTY_PRINT);
-    }
-
-    public function setTypemapAsJson($value): void
-    {
-        $this->typemap = Json::decode($value);
-    }
-
-    public function setOverridesAsJson(string $value): void
-    {
-        $this->overrides = array_filter(Json::decode($value));
-    }
-
     public function rules(): array
     {
         return [
-            [[
-                'title', 'base_survey_eid'
-            ], RequiredValidator::class],
-            [['title'], StringValidator::class],
-            [['title'], UniqueValidator::class],
-            [['base_survey_eid'], RangeValidator::class, 'range' => array_keys($this->dataSurveyOptions())],
+            [['title', 'base_survey_eid'], RequiredValidator::class],
+            [['title'], UniqueValidator::class, 'targetAttribute' => 'title'],
```

Comment:
Is the `targetAttribute` required? Doesn't the validator use the attribute itself for unique checking?